### PR TITLE
jupyter_bokeh 4.0.0

### PR DESCRIPTION
**Pull request by mattpap**
This is preparatory PR for jupyter_bokeh 3.0.8 release. The main change is support for jupyterlab 4 (Python side). Base on my testing, this soft of works, i.e. plots/layouts appear, however handles don't work and thus there's no interactivity.

### CODE REVIEW

**Comment by scimax** [comment]
Hi @mattpap , I just came across your PR because I experienced the same issue as stated in #195 . However, since you mention that the handles don't work anymore in your case, I was a bit confused.
I had a running instance of jupyterlab 4 still open while i was installing jupyter_bokeh 3.0.7. During the installation, jupyterlab was downgraded, but not the current instance, since it was still running. Then I imported bokeh in the open notebook and it worked. Doesn't that mean that theoretically jupyter_bokeh 3.0.7. is compatible with jupyterlab 4.x? Interactivity was working.
Or do I misinterpret something here in my observations?

**Comment by bryevdv** [comment]
@scimax "handles" refers specifically to the old `push_notebook` API, is that what you were using? "Bokeh app embedded in a notebook" is a different kind of usage that does not involve handles.

File: jupyter_bokeh/_version.py [hunk 1]
```diff
@@ -1,19 +1,4 @@
...
-        except FileNotFoundError:
-            pass
-
-    raise FileNotFoundError(f"Could not find package.json under dir {HERE!s}")
-
-__version__ = _fetchVersion()
-
+# This file is auto-generated by Hatchling. As such, do not:
+#   - modify
+#   - track in version control e.g. be sure to add to .gitignore
```
Comment (Hoxbro) [inline jupyter_bokeh/_version.py:3]: Just to make you aware of it.

Comment (philippjfr) [reply jupyter_bokeh/_version.py:3]: Ah, thanks. Will remove.

**Comment by philippjfr** [comment]
I've updated this PR to build against JupyterLab 4 now which included:

- Updating the package.json with latest Jupyter packages
- Replace `setup.py` and previous jupyter packaging approach with modern Jupyter tooling based on hatch(ling)
- Some typing changes.

File: scripts/deploy.sh [hunk 2]
```diff
@@ -4,11 +4,12 @@ set -e
...
 git clean -dfx
-npm install
+jlpm install
+jlpm run build:prod
 npm publish --access public # token is configured in ~/.npmrc; requires OTP
 
 git clean -dfx
-python setup.py sdist
+python setup.py sdist bdist_wheel
 twine upload -u __token__ -p $(cat ~/.tokens/pypi_jupyter_bokeh) dist/jupyter_bokeh-*.tar.gz
```
Comment (Hoxbro) [inline scripts/deploy.sh:13]: Can `hatch build` and `hatch publish` be used here?

**Comment by philippjfr** [comment]
@mattpap Would you mind testing and then releasing?

File: scripts/deploy.sh [hunk 3]
```diff
@@ -9,10 +9,12 @@ jlpm run build:prod
 npm publish --access public # token is configured in ~/.npmrc; requires OTP
 
 git clean -dfx
-python setup.py sdist bdist_wheel
-twine upload -u __token__ -p $(cat ~/.tokens/pypi_jupyter_bokeh) dist/jupyter_bokeh-*.tar.gz
+hatch build
+hatch publish -u __token__ -a $(cat ~/.tokens/pypi_jupyter_bokeh)
+
+VERSION=$(find dist -name "*.whl" -exec basename {} \; | cut -d- -f2)
```
Comment (Hoxbro) [inline scripts/deploy.sh:15]: Can be updated with `hatch version`

Comment (philippjfr) [reply scripts/deploy.sh:15]: Can you make the corresponding update in Panel?

Comment (Hoxbro) [reply scripts/deploy.sh:15]: Panel does not use hatch

Comment (philippjfr) [reply scripts/deploy.sh]: Duh!

**Comment by bryevdv** [comment]
Please just update the README with a few lines how to build/deploy using the new system

**Comment by philippjfr** [comment]
Nothing should have changed on that front, but @mattpap when you get a chance do document the deployment process.

**Pull request merged**